### PR TITLE
Output Liquid and Markdown Profiles in same table

### PR DESCRIPTION
**Pull request by ashmaroli**
Allow inspecting Liquid Profiling and Markdown Profiling in the same table

Sample Output:
```
                                                          | Liquid |   Liquid | Liquid | Markup |   Markup | Markup
Filename                                                  |  Count |    Bytes |   Time |  Count |    Bytes |   Time
----------------------------------------------------------+--------+----------+--------+--------+----------+-------
_layouts/default.html                                     |    125 | 3301.46K | 2.070s |      0 |    0.00K | 0.000s
_layouts/docs.html                                        |     49 | 1087.24K | 1.395s |      0 |    0.00K | 0.000s
_includes/docs_contents_mobile.html                       |     49 |  141.35K | 0.626s |      0 |    0.00K | 0.000s
_includes/docs_contents.html                              |     49 |  142.84K | 0.616s |      0 |    0.00K | 0.000s
_includes/docs_option.html                                |     49 |  113.55K | 0.562s |      0 |    0.00K | 0.000s
_includes/docs_ul.html                                    |     49 |  129.01K | 0.558s |      0 |    0.00K | 0.000s
_layouts/news.html                                        |     67 | 1164.76K | 0.368s |      0 |    0.00K | 0.000s
_includes/header.html                                     |    125 |  166.29K | 0.212s |      0 |    0.00K | 0.000s
_includes/news_contents.html                              |     67 |  226.16K | 0.165s |      0 |    0.00K | 0.000s
_includes/news_contents_mobile.html                       |     67 |  473.58K | 0.162s |      0 |    0.00K | 0.000s
_includes/news_item.html                                  |      2 |  293.52K | 0.144s |      0 |    0.00K | 0.000s
sitemap.xml                                               |      1 |   14.90K | 0.134s |      1 |   14.90K | 0.000s
_includes/top.html                                        |    126 |   16.76K | 0.133s |      0 |  369.44K | 1.486s
feed.xml                                                  |      1 |   41.62K | 0.099s |      1 |   41.62K | 0.000s
_layouts/news_item.html                                   |     65 |  159.10K | 0.095s |      0 |    0.00K | 0.000s
_includes/section_nav.html                                |     49 |   21.80K | 0.089s |      0 |    0.00K | 0.000s
_layouts/tutorials.html                                   |      6 |  123.85K | 0.085s |      0 |    0.00K | 0.000s
news/releases/index.html                                  |      1 |  136.23K | 0.084s |      1 |  136.23K | 0.000s
news/index.html                                           |      1 |  157.77K | 0.079s |      1 |  157.77K | 0.000s
_docs/history.md                                          |      1 |  223.20K | 0.075s |      2 |  589.08K | 1.098s
_includes/mobile-nav-items.html                           |    125 |   37.43K | 0.050s |      0 |    0.00K | 0.000s
_includes/primary-nav-items.html                          |    125 |   28.52K | 0.048s |      0 |    0.00K | 0.000s
_includes/analytics.html                                  |    126 |  117.26K | 0.045s |      0 |    0.00K | 0.000s
_includes/footer.html                                     |    125 |   68.24K | 0.029s |      0 |    0.00K | 0.000s
_includes/tutorials_contents_mobile.html                  |      6 |    4.41K | 0.015s |      0 |    0.00K | 0.000s
_includes/tutorials_contents.html                         |      6 |    3.84K | 0.013s |      0 |    0.00K | 0.000s
_layouts/error.html                                       |      1 |    5.30K | 0.012s |      0 |    0.00K | 0.000s
_posts/2014-03-27-jekyll-1-5-1-released.markdown          |      1 |    1.09K | 0.012s |      2 |    2.32K | 0.007s
_includes/tutorials_option.html                           |      6 |    2.84K | 0.010s |      0 |    0.00K | 0.000s
_includes/tutorials_ul.html                               |      6 |    3.19K | 0.009s |      0 |    0.00K | 0.000s
_docs/collections.md                                      |      1 |   11.78K | 0.009s |      2 |   43.96K | 0.046s
_includes/section_nav_tutorials.html                      |      6 |    2.52K | 0.009s |      0 |    0.00K | 0.000s
_includes/anchor_links.html                               |    126 |  127.97K | 0.008s |      0 |    0.00K | 0.000s
_includes/search/input.html                               |    125 |    9.03K | 0.006s |      0 |    0.00K | 0.000s
_includes/search/script.html                              |    125 |   45.65K | 0.006s |      0 |    0.00K | 0.000s
_tutorials/convert-existing-site-to-jekyll.md             |      1 |   23.21K | 0.005s |      2 |   77.18K | 0.102s
philosophy.md                                             |      1 |    2.20K | 0.004s |      2 |    4.84K | 0.003s
_tutorials/orderofinterpretation.md                       |      1 |    6.55K | 0.003s |      2 |   26.16K | 0.039s
_docs/permalinks.md                                       |      1 |   13.38K | 0.003s |      2 |   33.22K | 0.035s
_posts/2016-04-19-jekyll-3-0-4-released.markdown          |      1 |    1.04K | 0.002s |      2 |    2.36K | 0.004s
_posts/2016-10-06-jekyll-3-3-is-here.md                   |      1 |    5.04K | 0.002s |      2 |   13.41K | 0.022s
community/index.md                                        |      1 |    3.98K | 0.002s |      2 |   10.37K | 0.016s
_layouts/redirect.html                                    |      7 |    3.32K | 0.002s |      0 |    0.00K | 0.000s
_tutorials/navigation.md                                  |      1 |   18.82K | 0.002s |      2 |   98.86K | 0.102s
_docs/templates.md                                        |      1 |   18.96K | 0.002s |      2 |   52.89K | 0.056s
_includes/docs_version_badge.html                         |      5 |    0.48K | 0.001s |      0 |    0.00K | 0.000s
_posts/2014-03-27-jekyll-1-5-1-released.markdown/#excerpt |      1 |    0.10K | 0.001s |      2 |    0.31K | 0.002s
_docs/configuration.md                                    |      1 |   27.70K | 0.001s |      2 |   88.23K | 0.100s
_posts/2017-03-09-jekyll-3-4-2-released.markdown/#excerpt |      1 |    0.13K | 0.001s |      2 |    0.30K | 0.001s
_docs/datafiles.md                                        |      1 |    3.56K | 0.001s |      2 |   19.48K | 0.013s
----------------------------------------------------------+--------+----------+--------+--------+----------+-------
TOTAL                                                     |   1884 | 8732.54K | 8.062s |     36 | 1782.94K | 3.130s
```

~~**Note:** The table does not output a `TOTAL`  row..~~

### CODE REVIEW

**Comment by ashmaroli** [comment]
This forces every user to profile both Liquid and Markdown.
Question: Should this be implemented as an opt-in (e.g. `--profile-with-markdown`) instead, in case users prefer the older output from `--profile`?

**Comment by pathawks** [comment]
If I am running with `--profile` I presumably want as much information as is available.

File: lib/jekyll/site.rb [hunk 1]
```diff
@@ -12,7 +12,8 @@ class Site
                   :gems, :plugin_manager, :theme
 
     attr_accessor :converters, :generators, :reader
-    attr_reader   :regenerator, :liquid_renderer, :includes_load_paths
+    attr_reader   :regenerator, :liquid_renderer, :converter_profiler,
```
Comment (ashmaroli) [inline lib/jekyll/site.rb]: TODO: Undo this change before merge..

**Comment by DirtyF** [comment]
If we specify units for bytes, we should also specify units for time (seconds)

**Comment by ashmaroli** [comment]
The table now outputs a `TOTAL` row that shows the total of the associated column.
The key to implementing this row was inspired by https://github.com/jekyll/jekyll/commit/e42b37f2dd007a635fe470aa108278c80c830680 by @parkr

**Comment by parkr** [comment]
Let’s bring this back a bit. What is the user interested in? They want to know which document/layout/include is taking ages to render. That’s where users optimize. So they definitely want to know total render time for the document/layout. They might care about the Markdown/Liquid breakdown to help them understand where to optimize. I don’t think the bytes are immediately actionable, and maybe the top 10-15 documents/layouts represent the most actionable information (you just get a huge long tail otherwise). 

For more detailed information, we should provide granular timing/bytes data in JSON only if the user provides a flag. This can then be passed to other programs to interpret & aggregate however they wish.

**Comment by parkr** [comment]
I think we should create a Profiler class. I have a branch which does this and refactors the existing liquid stuff.

/cc @jekyll/performance about my above musings.

**Comment by ashmaroli** [comment]
> I have a branch which does this and refactors the existing liquid stuff.

@parkr How's the work on that branch coming along..? I think its best you open a PR and continue the work on it..
@mneumegen and myself agree that your work on the said branch is a nice attempt at enhancing the `--profile` output.. and could easily *trump* this PR itself..

**Comment by parkr** [comment]
I haven’t put more work into it since you opened this PR but i’ll pick it back up and open a PR soon!

**Comment by mattr-** [comment]
I want to come back to [@parkr's comment](https://github.com/jekyll/jekyll/pull/6620#issuecomment-354987788) for a bit. Time is definitely an important issue, but I don't think file/markup/rendered size necessarily is. He rightly points out that as a user, I care less about the size of the things as much as I do the time it takes to generate them.

What do you think about the proposal to create a JSON file that contains that detailed byte size information and leave it out of the table output? Would you be willing to shorten the table output here in favor of a later PR that also generates a JSON file?

**Comment by DirtyF** [comment]
closing in favor of #6689
